deprecate container, change docs to image_uri (#49752)
ray-project/ray: doc/source/serve/advanced-guides/multi-app-container.md

:::
::::

Then, build the corresponding images and push it to your choice of container registry. This tutorial uses `alice/whisper_image:latest` and `alice/resnet_image:latest` as placeholder names for the images, but make sure to swap out `alice` for a repo name of your choice.

::::{tab-set}
:::{tab-item} Whisper
```bash

> # Build the image from the Dockerfile using podman

Reviewer: ```suggestion
# Build the image from the Dockerfile using Podman
```

:::
:::{tab-item} Resnet
```bash

> # Build the image from the Dockerfile using podman

Reviewer: ```suggestion
# Build the image from the Dockerfile using Podman
```

:::
::::

Finally, you can specify the container image within which you want to run each application in the `image_uri` field of an application's runtime environment specification.

:::{note}

> Previously the feature could be accessed through the `container` field of the runtime environment. That API is now deprecated in favor of `image_uri`.

Reviewer: ```suggestion
Previously you could access the feature through the `container` field of the runtime environment. That API is now deprecated in favor of `image_uri`.
```

* **ERRO[0000] 'overlay' is not supported over overlayfs: backing file system is unsupported for this graph driver**
  * This error should only occur when you are running the Ray cluster inside a container. If you see this error when starting the replica actor, try volume mounting `/var/lib/containers` in the container that runs raylet. That is, add `-v /var/lib/containers:/var/lib/containers` to the command that starts the Docker container.
* **cannot clone: Operation not permitted; Error: cannot re-exec process**

>   * This error should only occur when you are running the Ray cluster inside a container. This error implies that you don't have the permissions to use Podman to start a container. You need to start the container that runs raylet, with privileged permissions by adding `--privileged`.

Reviewer: ```suggestion
  * This error should only occur when you're running the Ray cluster inside a container. This error implies that you don't have the permissions to use Podman to start a container. You need to start the container that runs raylet, with privileged permissions by adding `--privileged`.
```
